elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/dimension/EntityDimension.java

```diff
@@ -0,0 +1,218 @@
+import java.util.stream.Collectors;
+
+/**
+ * A dimension backed by a table.
+ * <p>
+ * {@link EntityDimension} is thread-safe and can be accessed by multiple threads.
+ */
+@Slf4j
+public class EntityDimension extends Column implements Dimension {
+
+    /**
+     * Returns the entity field that is defined to be a human displayable column of that entity.
+     *
+     * @param cls  The entity or a relation
+     * @param entityDictionary  A object that helps to find all fields with {@link FriendlyName} annotation
+     *
+     * @return friendlyName of the entity
+     *
+     * @throws IllegalStateException if more than 1 fields are annotated by the {@link FriendlyName}
+     */
+    public static String getFriendlyNameField(Class<?> cls, EntityDictionary entityDictionary) {
+        List<String> singleFriendlyName = entityDictionary.getAllFields(cls).stream()
+                .filter(field -> entityDictionary.attributeOrRelationAnnotationExists(
+                        cls,
+                        field,
+                        FriendlyName.class
+                ))
+                .collect(Collectors.toList());
+
+        if (singleFriendlyName.size() > 1) {
+            String message = String.format(
+                    "Multiple @FriendlyName fields found in entity '%s'. Can only have 0 or 1",
+                    cls.getName()
+            );
+            log.error(message);
+            throw new IllegalStateException(message);
+        }
+
+        return singleFriendlyName.isEmpty()
+                ? entityDictionary.getIdFieldName(cls) // no friendly name found; use @Id field as friendly name
+                : singleFriendlyName.get(0);
+    }
+
+    /**
+     * Returns the estimated cardinality of this a dimension field.
+     * <p>
+     * {@link #getDefaultCardinality() Default} is returned when the dimension is not annotated by {@link Cardinality}.
+     *
+     * @param dimension  The dimension field
+     * @param cls  The entity name of the dimension from which estimated cardinality is being looked up
+     * @param entityDictionary  An object that helps to retrieve {@link Cardinality} annotation from either entity or
+     * field
+     *
+     * @return cardinality of the dimension field
+     */
+    public static CardinalitySize getEstimatedCardinality(
+            String dimension,
+            Class<?> cls,
+            EntityDictionary entityDictionary
+    ) {
+        if (entityDictionary.isRelation(cls, dimension)) {
+            // try to get annotation from entity first
+            Cardinality annotation = entityDictionary.getAttributeOrRelationAnnotation(
+                    cls,
+                    Cardinality.class,
+                    dimension
+            );
+
+            if (annotation != null) {
+                return annotation.size();
+            }
+
+            // annotation is not on entity; then must be on field or method
+        }
+
+        Cardinality annotation = entityDictionary
```

Comment:
This is identical to the call on 83.  I think this is not what you intended.

Reply:
Ah yeah. Thanks!